metricbeat/module/windows/service/service_windows.go

```diff
@@ -135,6 +146,34 @@ func OpenService(handle ServiceDatabaseHandle, serviceName string, desiredAccess
 	return serviceHandle, nil
 }
 
+func QueryServiceConfig2(serviceHandle ServiceHandle, infoLevel ServiceConfigInformation) ([]byte, error) {
+	var serviceBufSize uint32
+	var serviceBytesNeeded uint32
+
+	if err := _QueryServiceConfig2(serviceHandle, infoLevel, nil, serviceBufSize, &serviceBytesNeeded); err != nil {
+		if ServiceErrno(err.(syscall.Errno)) != SERVICE_ERROR_INSUFFICIENT_BUFFER {
+			err := CloseServiceHandle(serviceHandle)
+			return nil, err
+		}
+		serviceBufSize += serviceBytesNeeded
+		buffer := make([]byte, serviceBufSize)
+
+		for {
+			if err := _QueryServiceConfig2(serviceHandle, infoLevel, &buffer[0], serviceBufSize, &serviceBytesNeeded); err != nil {
+				if ServiceErrno(err.(syscall.Errno)) != SERVICE_ERROR_INSUFFICIENT_BUFFER {
+					err := CloseServiceHandle(serviceHandle)
+					return nil, err
+				}
+				serviceBufSize += serviceBytesNeeded
```

Comment:
The buffer needs to be re-allocated to account for the increased size requirement. Otherwise this will be an infinite loop.